tesseract_kinematics/src/core/rep_inverse_kinematics.cpp

```diff
@@ -46,9 +46,58 @@ bool RobotWithExternalPositionerInvKin::update()
 {
   manip_inv_kin_->update();
   positioner_fwd_kin_->update();
-  return init(scene_graph_, manip_inv_kin_, manip_reach_, positioner_fwd_kin_, positioner_sample_resolution_, name_);
+  if (!init(scene_graph_, manip_inv_kin_, manip_reach_, positioner_fwd_kin_, positioner_sample_resolution_, name_))
+    return false;
+
+  if (sync_fwd_kin_ != nullptr)
+    synchronize(sync_fwd_kin_);
+
+  return true;
+}
+
+void RobotWithExternalPositionerInvKin::synchronize(ForwardKinematics::ConstPtr fwd_kin)
+{
+  if (numJoints() != fwd_kin->numJoints())
+    throw std::runtime_error("Tried to synchronize kinematics objects with different number of joints!");
+
+  if (!tesseract_common::isIdentical(orig_data_.joint_names, fwd_kin->getJointNames(), false))
+    throw std::runtime_error("Tried to synchronize kinematics objects with different joint names!");
+
+  if (!tesseract_common::isIdentical(orig_data_.link_names, fwd_kin->getLinkNames(), false))
+    throw std::runtime_error("Tried to synchronize kinematics objects with different link names!");
+
+  if (!tesseract_common::isIdentical(orig_data_.active_link_names, fwd_kin->getActiveLinkNames(), false))
```

Comment:
It passes the link names, but fails in the active link names. This is how the active links look like
Forward: 
```
positioner_link_1, positioner_base_link, world, base_link, link_1, link_2, link_3, link_4, link_5, link_6, flange, tool0
```
Inverse:
```
flange, link_1, link_2, link_3, link_4, link_5, link_6, positioner_link_1, positioner_link_2, positioner_tool0, tool0
```

Reply:
It looks like the forward kinematics list is incorrect. I now see why it is wrong because of how the kinematics chain is defined. I need to improve the way we are generating active link names. You can comment on this check which should not affect the motion planners. It will put a few static links in the active tree but they will never move during collision checking. I will update the unit tests to add this kinematic group to develop a better way for getting active links.

Reply:
I just pushed a commit that should fix things.

Reply:
I Commented the active link check as it still not working. If it may be helpful here is how the active links look like
Forward:
```
positioner_link_1, link_1, link_2, link_3, link_4, link_5, link_6, flange, tool0
```
Inverse:
```
flange, link_1, link_2, link_3, link_4, link_5, link_6, positioner_link_1, positioner_link_2, positioner_tool0, tool0
```

Reply:
Thank you. I will keep working on things. I should be able to write some unit tests this weekend to find the edge cases. I

Is the Inverse kinematics the REP inverse kinematics object?

Reply:
it is the ` orig_data_.active_link_names`. I noticed that you also log these data in the terminal

Reply:
Alright, I have updated the generation of the active links within ROP and REP which I think should resolve the issue. Also fixed another issue in kdl generation of active links when the base is actually a child of a joint. Let me know if this changes anything on your end.

Reply:
It still fails on the active link condition
Forward (fwd_kin->getActiveLinkNames()):
```
positioner_tool0, positioner_link_1, link_1, link_2, link_3, link_4, link_5, link_6, flange, tool0,
```
Inverse (orig_data_.active_link_names):
```
flange, link_1, link_2, link_3, link_4, link_5, link_6, positioner_link_1, positioner_link_2, positioner_tool0, tool0,
```

Reply:
Now the forward looks correct, but there is an extra link in the inverse. Is the urdf public where I can take a look?

Reply:
> Now the forward looks correct, but there is an extra link in the inverse. Is the urdf public where I can take a look?

Yes, I'm using the positioner from [tesseract_ros_workcell](https://github.com/mohamedsayed18/motion_planner/blob/main/tesseract_ros_workcell/twc_support/CAD/robot/abb_irbpa250_d1000/urdf/irbpa250_d1000_macro.xacro)
and the abb_robot from [here](https://github.com/mohamedsayed18/motion_planner/blob/main/abb_irb2600_support/urdf/irb2600_12_165_macro.xacro)

Reply:
Ok I will look into it.

Reply:
it still fails in the active links
fwd_kin->getActiveLinkNames():
```
positioner_link_1, link_1, link_2, link_3, link_4, link_5, link_6, flange, tool0`
```
orig_data_.active_link_names
```
flange, link_1, link_2, link_3, link_4, link_5, link_6, positioner_link_1, positioner_link_2, positioner_tool0, tool0,
```
I commented it for now. 
Does this error affect the synchronization?